### PR TITLE
Update -enable-ossa-modules to be a module interface ignoreable option

### PR DESCRIPTION
**Pull request by meg-gupta**
This lets the flag to be emitted in the swiftinterface file of the stdlib (or other modules with -enable-ossa-modules specified), so that whenever stdlib is recompiled, -enable-ossa-modules will be on.

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci test

File: lib/Serialization/ModuleFormat.h [hunk 1]
```diff
@@ -58,7 +58,7 @@ const uint16_t SWIFTMODULE_VERSION_MAJOR = 0;
 /// describe what change you made. The content of this comment isn't important;
 /// it just ensures a conflict if two people change the module format.
 /// Don't worry about adhering to the 80-column limit for this line.
-const uint16_t SWIFTMODULE_VERSION_MINOR = 820; // begin_borrow/move_value [var_decl]
+const uint16_t SWIFTMODULE_VERSION_MINOR = 821; // add -enable-ossa-modules to .swiftinterface file
```
Comment (tshortli) [inline lib/Serialization/ModuleFormat.h:61]: Why does this require a module format version change? If the compiler already supports serializing and deserializing binary modules OSSA information either present or not present, then this change shouldn't have a meaningful effect on deserialization.

Comment (meg-gupta) [reply lib/Serialization/ModuleFormat.h:61]: Okay, I can remove it. I thought changing the stdlib's .swiftinterface in any way also needed a version update.

Comment (tshortli) [reply lib/Serialization/ModuleFormat.h:61]: It's of course fairly harmless to bump the module version, but I wanted to understand if there was something I was missing. I don't know of any reason that changes to the stdlib interface in general would require a module version bump.

**Review by tshortli — APPROVED** [review]

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
Thanks to @nate-chandler for helping me find `ModuleInterfaceOptionIgnorable`!

**Review by nate-chandler — APPROVED** [review]

**Pull request merged**